lib/location/method_gnss.c

```diff
@@ -739,6 +800,11 @@ int method_gnss_location_get(const struct location_method_config *config)
 {
 	int err;
 
+	err = check_modem_supports_gnss();
+	if (err) {
+		return err;
```

Comment:
Maybe it would be good to have a warning or error trace here as well. If this check fails, the current system mode doesn't match the Location library method configuration, so one of them is incorrect.